pkg/crc/preflight/preflight_linux.go

```diff
@@ -118,9 +93,29 @@ func getPreflightChecks() []Check {
 	checks = append(checks, nonWinPreflightChecks[:]...)
 	checks = append(checks, libvirtPreflightChecks[:]...)
 
+	distro, _ := getLinuxDistroInfo()
+	switch distro {
+	case "fedora", "centos", "rhel":
+		checks = append(checks, redhatPreflightChecks[:]...)
+	default:
+		logging.Warnf("dnsmasq related changes are not implemented for %s", distro)
+	}
 	return checks
 }
 
 func CleanUpHost() error {
 	return doCleanUpPreflightChecks(getPreflightChecks())
 }
+
+func getLinuxDistroInfo() (string, error) {
+	// Read the `/etc/os-release info from the host
+	content, err := ioutil.ReadFile("/etc/os-release")
+	if err != nil {
+		return "", err
+	}
+	osRelease, err := crcos.GetOsRelease(content)
+	if err != nil {
+		return "", err
+	}
+	return osRelease.ID, nil
```

Comment:
I'd move this to a `GetOsReleaseFile` method in ` pkg/os/linux/release_info.go `, the preflight package has no need to know that os release info is in `/etc/os-release`.
There could be a `getDistroSpecificPreflights()` helper though.